src/ws.c

```diff
@@ -245,6 +265,14 @@ static void* ws_establishconnection(void *vsock)
 	}
 
 closed:
+	/* Removes client socket from socks list. */
+	pthread_mutex_lock(&mutex);
+	for (unsigned int i = 0; i < MAX_CLIENTS; i++)
+	{
+		if (client_socks[i] == sock)
+			client_socks[i] = 0;
```

Comment:
Although hard to happen, the docs states that `accept()` must return a non-negative number for the accepted socket, which also includes 0 (even though 0 is guaranteed to be `stdin`, just to be safe here).

So what I suggest to you is change your 'empty-slot' value to -1, since it is guaranteed to be an invalid value for a file descriptor. You can also break the loop as soon as found the corresponding fd value, something like:
```c
if (client_socks[i] == sock)
    client_socks[i] = -1, i = MAX_CLIENTS;
```